src/modules/user/user.controller.ts

```diff
@@ -45,4 +57,32 @@ export class UserController {
       message: result.message,
     };
   }
+
+  @ApiOperation({ summary: 'Get user' })
+  @ApiResponse({
+    status: 200,
+    description: 'Get user by Id',
+    type: GetUserByIdResponseDto,
+  })
+  @Get(':userId')
+  async getUserById(@Param() { userId }: { userId: string }) {
+    try {
+      const { password, ...userData } = await this.userService.getUserRecord({
+        identifier: userId,
+        identifierType: 'id',
+      });
+      return {
+        status_code: 200,
+        user: userData,
+      };
+    } catch (e) {
+      throw new HttpException(
+        {
+          status_code: 500,
+          error: 'Internal Server Error',
+        },
```

Comment:
why are you using only controller? This code should be in service file

Reply:
the service was already designed. I can't temper with it